### ISSUE TITLE
Soroban Rpc: effective db migrations at startup

### ISSUE THREAD

**Issue by sreuland** [stellar/soroban-tools#991]
### What problem does your feature solve?
after upgrading rpc to a newer version that has different db schema, have observed the restart against prior db fails with schema errors. Have run into this on restart of rpc on kubernetes deployments of the rpc image, since it stores the db on a PersistentVolume, in order to have dedicated/stable access to known storage and limits rather than hoping it will 'fit' on default ephemeral pod storage, which is variant.

### What would you like to see?
rpc start-up recognizes older db schemas if present in the db instance provided at runtime and attempts migrations where possible to lift the given db to current schema required, allows rpc to have a successful, autonomous start-up after a sw version upgrade containing a db schema change 

### What alternatives are there?
before restarting rpc on new sw version with new schema, first delete the prior db it had created and then start rpc server, it will create a new db from scratch.

**Comment by sreuland** [comment]
@2opremio , this bug report from a couple preview releases ago seems similar to  behavior observed today during the reset, upon restart of new rpc version in a new container, assuming new/empty storage established, observed startup error:

```
could not obtain txmeta cache from the database" error="sql: Scan error on column index 0, name \"meta\": union LedgerCloseMeta has invalid V (int32) switch value '2'

```

**Comment by 2opremio** [comment]
Yeah, the XDR changed but the db wasn’t reset (for some reason it still contained old data)

**Comment by sreuland** [comment]
i think this can be closed as obsolete, it's not occuring or replicatable, and xdr changes are done.
